### PR TITLE
basic stats export

### PR DESCRIPTION
**Pull request by AleksanderSkubala**
Hi guys, I made this basic stats export. I'm aware of the fact, that it could look a bit hardcoded, but I tried to do it as quickly as possible 💕 Let me know what you think about it

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,14 +28,16 @@
     "@types/faker": "5.5.9",
     "axios": "0.24.0",
     "bcrypt": "5.0.1",
+    "excel4node": "1.7.2",
     "faker": "5.5.3",
+    "js-file-download": "0.4.12",
     "lodash.debounce": "^4.0.8",
```
Comment (AdamSiekierski) [inline package.json:34]: ummmm... let's remove that `^`

File: pages/admin/dashboard.tsx [hunk 2]
```diff
@@ -3,9 +3,22 @@ import { withServerSideAuth } from "../../lib/auth/withServerSideAuth";
...
 const AdminPage = ({ user }: InferGetServerSidePropsType<typeof getServerSideProps>) => {
   const theme = useTheme();
+  const downloadStats = () => {
+    axios({
+      url: "/api/exportStats",
+      method: "GET",
+      responseType: "blob",
+    })
+      .then((response) => fileDownload(response.data, `zsk-amnesty.xlsx`))
+      .catch((err) => console.log(err));
```
Comment (AdamSiekierski) [inline pages/admin/dashboard.tsx:19]: ah yes, error handling 👀 we do have our toast component for a reason 😅

File: pages/api/exportStats.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+export default withApiAuth(
+  withApiMethods({
+    GET: async (_req, res) => {
+      try {
+        const allLetters = await prisma.letter.findMany();
+        const allAuthors = await prisma.author.findMany();
+        const allDestinations = await prisma.destination.findMany();
+
+        console.log(allAuthors, allDestinations);
```
Comment (AdamSiekierski) [inline pages/api/exportStats.ts:14]: rogue console log 👁️👁️

File: pages/api/exportStats.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        const normalStyle = workbook.createStyle({
+          font: {
+            bold: false,
+            size: 12,
+          },
+        });
+
+        worksheet1.cell(1, 1).string("Class name").style(headingStyle);
+        worksheet1.cell(1, 2).string("Register number").style(headingStyle);
+        worksheet1.cell(1, 3).string("Destination").style(headingStyle);
```
Comment (AdamSiekierski) [inline pages/api/exportStats.ts:34]: I'd separate the whole stats exporting logic to a separate file tho. And to some separate functions.

The `/lib/statsExport` would be a perfect home for this code

File: pages/api/exportStats.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+          const destination = allDestinations.find(
+            (destination) => destination.id === letter.destinationId,
+          );
+
+          worksheet1.cell(i, 1).string(author?.classId).style(normalStyle);
+          worksheet1.cell(i, 2).number(author?.registerNumber).style(normalStyle);
+          worksheet1.cell(i, 3).string(destination?.name).style(normalStyle);
+
+          i++;
+        });
```
Comment (AdamSiekierski) [inline pages/api/exportStats.ts:48]: I don't think using a forEach loop is a perfect thing to do. Especially with that iterator variable xD

**Review by AdamSiekierski — CHANGES_REQUESTED** [review]

File: pages/api/exportStats.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
+import { withApiAuth } from "../../lib/auth/withApiAuth";
+import { withApiMethods } from "../../lib/withApiMethods";
+import { prisma } from "../../lib/db";
+const excel = require("excel4node");
```
Comment (AdamSiekierski) [inline pages/api/exportStats.ts:4]: Too much [Require Podcast](https://require.podcast.gq)?

File: pages/api/exportStats.ts (hunk 3 above)
Comment (AleksanderSkubala) [reply pages/api/exportStats.ts:14]: ahh yes development reasons

File: pages/api/exportStats.ts (hunk 6 above)
Comment (datejer) [reply pages/api/exportStats.ts:4]: image

**Comment by AleksanderSkubala** [comment]
🙏🏻

File: pages/api/exportStats.ts (hunk 6 above)
Comment (AleksanderSkubala) [reply pages/api/exportStats.ts]: yes, es6 import is indeed joke to me 😈

File: pages/admin/dashboard.tsx [hunk 7]
```diff
@@ -3,9 +3,24 @@ import { withServerSideAuth } from "../../lib/auth/withServerSideAuth";
...
 const AdminPage = ({ user }: InferGetServerSidePropsType<typeof getServerSideProps>) => {
   const theme = useTheme();
+  const [error, setError] = useState("");
+  const downloadStats = () => {
+    axios({
+      url: "/api/exportStats",
+      method: "GET",
+      responseType: "blob",
+    })
+      .then((response) => fileDownload(response.data, `zsk-amnesty.xlsx`))
```
Comment (AdamSiekierski) [inline pages/admin/dashboard.tsx:20]: Any browser having some kind of popup protection would block this behaviour

File: types/excel4node/index.d.ts [hunk 8]
```diff
@@ -0,0 +1 @@
+declare module "excel4node";
```
Comment (AdamSiekierski) [inline types/excel4node/index.d.ts:1]: add an EOL to the end of this file

**Review by AdamSiekierski — CHANGES_REQUESTED** [review]
mostly small things - when you get them resolved, we're ready to 🚀

File: .eslintrc.json [hunk 9]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": ["next/core-web-vitals", "plugin:prettier/recommended"],
   "rules": {
-    "@next/next/no-img-element": 0
+    "@next/next/no-img-element": 0,
+    "@next/next/no-html-link-for-pages": 0
```
Comment (AdamSiekierski) [inline .eslintrc.json:5]: well, generally I wouldn't disable this rule for the *whole* project, and ignore it at the line level.

**Comment by AleksanderSkubala** [comment]
👉🏻👈🏻

**Review by AdamSiekierski — APPROVED** [review]
greyt job, lgtm 🚀🥵

**Comment by AleksanderSkubala** [comment]
> greyt job, lgtm 🚀🥵

Les go <3

**Pull request merged**